spec/controllers/tournaments_controller_spec.rb

```diff
@@ -282,6 +312,103 @@
           expect(response).to have_http_status(:ok)
           expect(response.content_type).to eq('application/json')
         end
+
+        context 'any variable relevant for group stage to playoff transition changed' do
+          before do
+          end
+
+          before(:each) do
+            @filled_tournament = create(:group_stage_tournament)
+            apply_authentication_headers_for @filled_tournament.owner
+          end
+
+          it 'fails when only instant_finalists_amount is changed' do
+            put :update, params: { id: @filled_tournament.to_param }.merge(instant_finalists_amount: 29)
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
+
+          it 'fails when only intermediate_round_participants_amount is changed' do
+            put :update, params: { id: @filled_tournament.to_param }.merge(intermediate_round_participants_amount: 29)
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
+
+          it 'fails when parameters don\'t match' do
+            put :update, params: { id: @filled_tournament.to_param }.merge(intermediate_round_participants_amount: 29,
+                                                                           instant_finalists_amount: 32)
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
+
+          it 'succeeds when all three are changed correctly' do
+            put :update, params: { id: @filled_tournament.to_param }.merge(intermediate_round_participants_amount: 2,
+                                                                           instant_finalists_amount: 1,
+                                                                           playoff_teams_amount: 2)
+          end
+
+          context 'only playoff_teams_amount is changed reasonably but update fails' do
+            before do
+              allow_any_instance_of(Tournament)
+                .to receive(:update)
+                .and_return(false)
+            end
+
+            it 'returns unprocessable entity' do
+              put :update, params: { id: @filled_tournament.to_param }.merge(playoff_teams_amount: 8)
+              expect(response).to have_http_status(:unprocessable_entity)
+            end
+
+            it 'doesn\'t change playoff_teams_amount' do
+              expect do
+                put :update, params: { id: @filled_tournament.to_param }.merge(playoff_teams_amount: 8)
+                @filled_tournament.reload
+              end
+                .to_not(change { @filled_tournament.playoff_teams_amount })
+            end
+
+            it 'doesn\'t change instant_finalists_amount' do
+              expect do
+                put :update, params: { id: @filled_tournament.to_param }.merge(playoff_teams_amount: 8)
+                @filled_tournament.reload
+              end
+                .to_not(change { @filled_tournament.instant_finalists_amount })
```

Comment:
this probably belongs into the line above

Reply:
I put it down there because I found it hard to read, was pretty cluttered

Reply:
Looks weird to me having it hang around there, but if you think it improves readability, I'm fine with that

Reply:
it does look weird, yes, but I would leave it like that if it is okay for you.. wont kill me if I change it to be in one line :D